distributed/worker_state_machine.py

```diff
@@ -372,6 +425,16 @@ class ExecuteSuccessEvent(StateMachineEvent):
     type: type | None
     __slots__ = tuple(__annotations__)  # type: ignore
 
+    def log(self, *, handled: float) -> StateMachineEvent:
+        out = copy(self)
+        out.handled = handled
+        out.value = None
```

Comment:
I understand the execution result is discarded because of the potentially large size of the result, and possibly the complexity of serialising/deserialising the result?

Reply:
Not discarding it would cause worker.stimulus_log to become effecitvely a copy of worker.data, except that it never loses any data!

Reply:
Indeed!